src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -145,22 +166,22 @@ class ReferralService(
     return referralAccessFilter.serviceProviderReferralSummaries(referralSummaries, user)
   }
 
-  private fun getSentReferralsForProbationPractitionerUser(user: AuthUser): List<Referral> {
-    val referralsStartedByPP = referralRepository.findByCreatedByAndSentAtIsNotNull(user)
-    val referralsManagedByPP = try {
-      communityAPIOffenderService.getManagedOffendersForDeliusUser(user)
+  private fun getSentReferralsForProbationPractitionerUser(user: AuthUser, filterSpecification: Specification<Referral>): List<Referral> {
+    var referralsForPPUser = ReferralSpecifications.createdBy(user)
+    try {
+      val serviceUserCRNs = communityAPIOffenderService.getManagedOffendersForDeliusUser(user).map { it.crnNumber }
+      referralsForPPUser = referralsForPPUser.or(ReferralSpecifications.matchingServiceUserReferrals(serviceUserCRNs))
```

Comment:
i would prefer to take this logic out of the try catch block - the catch does not handle errors in this code, only in the call to the community API.

Reply:
Understood. Was not familiar with transforming try/catch into an anonymous function block.